### PR TITLE
Add Cluster API v2 doc to complement its reference page

### PR DESCRIPTION
**Pull request by mikeCRL**
This is an enhanced version of the doc originally reviewed by @piyush-singh and @itsbilal. It still needs a thorough review for technical accuracy as it has had many updates.

Preview:  
https://deploy-preview-10532--cockroachdb-docs.netlify.app/docs/v21.1/cluster-api  
https://deploy-preview-10532--cockroachdb-docs.netlify.app/docs/v21.1/monitoring-and-alerting

Please also see the linked [API reference](https://cockroachlabs.com/docs/api/cluster/v2.html). Further enhancements to that page will come in separate PRs, but recommendations are welcome. Intended improvements include the following, though not all may be completed prior to the CRDB v21.1 release:

* Template/Redoc settings
   - Limit request sample languages: perhaps just to shell+curl, maybe a couple others in addition.
   - Remove the Powered by Redoc link in the left nav; maybe replace with link to 'edit this page' and move redoc link to bottom of main column.
   - Tweaking other redoc [options](https://github.com/Redocly/redoc#redoc-options-object).
* Spec
   - Add some remaining missing lower-level parameters.
   - Confirm that parameters described as internal should be so.
   - Refine parameter descriptions.
   - Make the `https` URLs for API endpoints the defaults in the drop-down and code samples, if possible, instead of the `http` URLs.
   - Add `--cacert` to curl calls since this the API will almost always be used for secure clusters.
   - Add custom example JSON responses where especially helpful.
   - Link back to this API doc (cluster-api.md).
 * Other
   - Troubleshoot {range_id} and {node_id} showing up in URLs in examples as %7Brange_id%7D and %7Bnode_id%7D. (Possibly implement a manual spec edit, but if this is coming from one of the scripts to transform the spec, we could follow them by a sed command in the build script to find/replace.)

### CODE REVIEW

File: v21.1/cluster-api.md
Comment (knz) [inline v21.1/cluster-api.md:32]: I don't think that's accurate - the TLS cert is only needed to create the session token via the CLI Command, but then it's also not the only way, the HTTP client can also go to the `/login` endpoint and use that to generate a session token.

Once they have a session token, that's the only thing they need to access the API.

**Comment by mikeCRL** [comment]
@taroface Would you mind reviewing while I attempt to resolve Raphael's correction?

**Review by taroface — APPROVED** [review]
LGTM with some suggestions, and the generated API docs look really slick, nice one!

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/docs/10532#-:-M_XtYcUBEWSsHeYwJU-:byjyx3m)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @mikeCRL)

___
*[v21.1/cluster-api.md, line 9 at r7](https://reviewable.io/reviews/cockroachdb/docs/10532#-M_XqIAX78UUl6EvKeGK:-M_XqIAX78UUl6EvKeGL:bpmgn2h) ([raw file](https://github.com/cockroachdb/docs/blob/e36084023d4e34a43e4e99b3d7d14b1c345f4782/v21.1/cluster-api.md#L9)):*
> ```Markdown
> The CockroachDB Cluster API is a REST API that provides information about a cluster and its nodes. The API offers programmatic access to much of the information available in the [DB Console](ui-overview.html) user interface, enabling you to monitor and troubleshoot your cluster using your choice of tooling.
> 
> The Cluster API is hosted by all nodes of your cluster and provides information about all nodes. The API is available on the same port that is listening for HTTP connections to the DB Console.
> ```

Suggestion: "the same port that listens for HTTP connections"

Also, mention default port (I think it's `8080`)?
___
*[v21.1/cluster-api.md, line 20 at r7](https://reviewable.io/reviews/cockroachdb/docs/10532#-M_XqqhY5WBr_n4cxgh9:-M_XqqhY5WBr_n4cxghA:b-clgnq7) ([raw file](https://github.com/cockroachdb/docs/blob/e36084023d4e34a43e4e99b3d7d14b1c345f4782/v21.1/cluster-api.md#L20)):*
> ```Markdown
> --- | --- | ---
> [`/health`](../api/cluster/v2.html#operation/health) | Check node health | Determine if the node is running and ready to accept SQL connections.
> [`/nodes`](../api/cluster/v2.html#operation/listNodes) | List nodes | Get information on all nodes in the cluster including node IDs, software versions, and hardware.
> ```

Suggestion: "Get details on all nodes in the cluster, including..."
___
*[v21.1/cluster-api.md, line 21 at r7](https://reviewable.io/reviews/cockroachdb/docs/10532#-M_XtVH54Yb4VUCbZc_s:-M_XtVH54Yb4VUCbZc_t:b-bfy1qe) ([raw file](https://github.com/cockroachdb/docs/blob/e36084023d4e34a43e4e99b3d7d14b1c345f4782/v21.1/cluster-api.md#L21)):*
> ```Markdown
> [`/health`](../api/cluster/v2.html#operation/health) | Check node health | Determine if the node is running and ready to accept SQL connections.
> [`/nodes`](../api/cluster/v2.html#operation/listNodes) | List nodes | Get information on all nodes in the cluster including node IDs, software versions, and hardware.
> [`/nodes/{node_id}/ranges`](../api/cluster/v2.html#operation/listNodeRanges) | List node ranges | For a specified node, obtain details on the ranges that it hosts. 
> ```

Suggest "obtain" -> "get" for consistency with the other API descriptions
___
*[v21.1/cluster-api.md, line 22 at r7](https://reviewable.io/reviews/cockroachdb/docs/10532#-M_Xr5TC4psra7zUpIP9:-M_Xr5TC4psra7zUpIPA:b-ecxpp0) ([raw file](https://github.com/cockroachdb/docs/blob/e36084023d4e34a43e4e99b3d7d14b1c345f4782/v21.1/cluster-api.md#L22)):*
> ```Markdown
> [`/nodes`](../api/cluster/v2.html#operation/listNodes) | List nodes | Get information on all nodes in the cluster including node IDs, software versions, and hardware.
> [`/nodes/{node_id}/ranges`](../api/cluster/v2.html#operation/listNodeRanges) | List node ranges | For a specified node, obtain details on the ranges that it hosts. 
> [`/ranges/hot`](../api/cluster/v2.html#operation/listHotRanges) | List hot ranges | Get information on ranges receiving a high number of reads or writes.
> ```

Suggest "Get details" for consistency with the other API descriptions

Nit: Ideally we could have a more specific threshold for "high number". Not sure if there is a concrete definition of "hot ranges" though.
___
*[v21.1/cluster-api.md, line 23 at r7](https://reviewable.io/reviews/cockroachdb/docs/10532#-M_XrL4u8FR2200KGUbU:-M_XrL4u8FR2200KGUbV:btu4fs5) ([raw file](https://github.com/cockroachdb/docs/blob/e36084023d4e34a43e4e99b3d7d14b1c345f4782/v21.1/cluster-api.md#L23)):*
> ```Markdown
> [`/nodes/{node_id}/ranges`](../api/cluster/v2.html#operation/listNodeRanges) | List node ranges | For a specified node, obtain details on the ranges that it hosts. 
> [`/ranges/hot`](../api/cluster/v2.html#operation/listHotRanges) | List hot ranges | Get information on ranges receiving a high number of reads or writes.
> [`/ranges/{range_id}`](../api/cluster/v2.html#operation/listRange) | Get range details | Get detailed technical information on a range. Typically of use only to Cockroach Labs engineers.
> ```

Suggestion: "Typically used by Cockroach Labs engineers."
___
*[v21.1/cluster-api.md, line 24 at r7](https://reviewable.io/reviews/cockroachdb/docs/10532#-M_XrbXwALoU_nqo6ZoV:-M_XrbXwALoU_nqo6ZoW:bj8ev8b) ([raw file](https://github.com/cockroachdb/docs/blob/e36084023d4e34a43e4e99b3d7d14b1c345f4782/v21.1/cluster-api.md#L24)):*
> ```Markdown
> [`/ranges/hot`](../api/cluster/v2.html#operation/listHotRanges) | List hot ranges | Get information on ranges receiving a high number of reads or writes.
> [`/ranges/{range_id}`](../api/cluster/v2.html#operation/listRange) | Get range details | Get detailed technical information on a range. Typically of use only to Cockroach Labs engineers.
> [`/sessions`](../api/cluster/v2.html#operation/listSessions) | List sessions | Get details of all current users' SQL sessions or sessions by a specific user.
> ```

Suggestion: "Get SQL session details of all current users or a specified user."
___
*[v21.1/cluster-api.md, line 25 at r7](https://reviewable.io/reviews/cockroachdb/docs/10532#-M_XrmivDIGUnWaIdNxM:-M_XrmivDIGUnWaIdNxN:buwuh31) ([raw file](https://github.com/cockroachdb/docs/blob/e36084023d4e34a43e4e99b3d7d14b1c345f4782/v21.1/cluster-api.md#L25)):*
> ```Markdown
> [`/ranges/{range_id}`](../api/cluster/v2.html#operation/listRange) | Get range details | Get detailed technical information on a range. Typically of use only to Cockroach Labs engineers.
> [`/sessions`](../api/cluster/v2.html#operation/listSessions) | List sessions | Get details of all current users' SQL sessions or sessions by a specific user.
> [`/login`](../api/cluster/v2.html#operation/login) | Log in | Authenticate as an admin on the cluster to retrieve a session token to use with further API calls.
> ```

Should this be `admin`, as in `admin` role? https://www.cockroachlabs.com/docs/v20.2/authorization.html#default-roles
___
*[v21.1/cluster-api.md, line 34 at r7](https://reviewable.io/reviews/cockroachdb/docs/10532#-M_Xs21-7HPOUCYrMHu9:-M_Xs21-7HPOUCYrMHuA:b-c1bbj7) ([raw file](https://github.com/cockroachdb/docs/blob/e36084023d4e34a43e4e99b3d7d14b1c345f4782/v21.1/cluster-api.md#L34)):*
> ```Markdown
> * A [client TLS certificate generated for the SQL role](cockroach-cert.html#create-the-certificate-and-key-pair-for-a-client), using the same CA cert used by the cluster. 
> 
> All endpoints except `health/` and `login/` require authentication using a session token. To obtain a session token, you will need:
> ```

Should these be `/health` and `/login` or is this a convention I'm unfamiliar with?
___
*[v21.1/cluster-api.md, line 36 at r7](https://reviewable.io/reviews/cockroachdb/docs/10532#-M_Xs5xN2edWm7n1N8IJ:-M_Xs5xN2edWm7n1N8IK:bks3r7) ([raw file](https://github.com/cockroachdb/docs/blob/e36084023d4e34a43e4e99b3d7d14b1c345f4782/v21.1/cluster-api.md#L36)):*
> ```Markdown
> All endpoints except `health/` and `login/` require authentication using a session token. To obtain a session token, you will need:
> 
> * A [SQL role](create-role.html) that is a member of the admin role and has login permissions and a password. You will use these credentials with the `login/` endpoint to retrieve the session token which you can then use with further API cals.
> ```

Use backticks on "admin"
___
*[v21.1/cluster-api.md, line 40 at r7](https://reviewable.io/reviews/cockroachdb/docs/10532#-M_XsCNI5g_XGE8W559I:-M_XsCNI5g_XGE8W559J:b-w5gt8s) ([raw file](https://github.com/cockroachdb/docs/blob/e36084023d4e34a43e4e99b3d7d14b1c345f4782/v21.1/cluster-api.md#L40)):*
> ```Markdown
> ## Authentication
> 
> 1. Retrieve a session token using the `login/` endpoint. For example:
> ```

`/login`?
___
*[v21.1/cluster-api.md, line 42 at r7](https://reviewable.io/reviews/cockroachdb/docs/10532#-M_XsS39093Fr3JKqYSP:-M_XsS39093Fr3JKqYSQ:b9bs5ao) ([raw file](https://github.com/cockroachdb/docs/blob/e36084023d4e34a43e4e99b3d7d14b1c345f4782/v21.1/cluster-api.md#L42)):*
> ```Markdown
> 1. Retrieve a session token using the `login/` endpoint. For example:
> 
>    ``` shell
> ```

Include `{% include copy-clipboard.html %}` to add the Copy button on any example commands.

Also, replace triple backticks with `~~~` throughout.
___
*[v21.1/cluster-api.md, line 58 at r7](https://reviewable.io/reviews/cockroachdb/docs/10532#-M_Xsizr5Sr8PTsgNciR:-M_Xsizr5Sr8PTsgNciS:bwti93e) ([raw file](https://github.com/cockroachdb/docs/blob/e36084023d4e34a43e4e99b3d7d14b1c345f4782/v21.1/cluster-api.md#L58)):*
> ```Markdown
> 
>    ``` shell
>    curl -H "X-Cockroach-API-Session: <token>" \
> ```

Since this is an example, maybe `<token>` should go ahead and use the actual token string?

If not, perhaps replace `< >` with `{ }` for style consistency.
___
*[v21.1/cluster-api.md, line 65 at r7](https://reviewable.io/reviews/cockroachdb/docs/10532#-M_Xsxhh-pRP1jX_MKB-:-M_Xsxhh-pRP1jX_MKB0:bker6p4) ([raw file](https://github.com/cockroachdb/docs/blob/e36084023d4e34a43e4e99b3d7d14b1c345f4782/v21.1/cluster-api.md#L65)):*
> ```Markdown
> ## Versioning and Stability
> 
> Future versions of CockroachDB will continue to provide access to the v2.0 API until it is deprecated and may conconurrently provide multiple API versions.
> ```

Suggest breaking into two sentences. 

"...until it is deprecated. Multiple API versions may be provided concurrently." (note typo in "concurrently")
___
*[v21.1/cluster-api.md, line 67 at r7](https://reviewable.io/reviews/cockroachdb/docs/10532#-M_Xt92o7NySpdKBkguM:-M_Xt92o7NySpdKBkguN:bu7pdcn) ([raw file](https://github.com/cockroachdb/docs/blob/e36084023d4e34a43e4e99b3d7d14b1c345f4782/v21.1/cluster-api.md#L67)):*
> ```Markdown
> Future versions of CockroachDB will continue to provide access to the v2.0 API until it is deprecated and may conconurrently provide multiple API versions.
> 
> All endpoint paths and payloads will remain available within a major API version number (v2.x). Minor versions could add new endpoints but would not remove existing endpoints.
> ```

Suggestion: "would not" -> "will not"

**Comment by jseldess** [comment]
@mikeCRL, how are you planning to expose this content in the docs sidenav?

**Comment by mikeCRL** [comment]
> @mikeCRL, how are you planning to expose this content in the docs sidenav?

Thanks, @jseldess. I've been meaning to add this to the PR. The sidebar-data JSON now has this structure:

CRDB > **Deploy > Monitoring and Alerting > Cluster API** (`v21.1/cluster-api.html` - the main/conceptual doc)  
CRDB > **Reference > Cluster API** (`api/cluster/v2.html` - the reference doc)

(The new CI-check failure seems unrelated; probably due to my rebasing. Suddenly, links to the subsections of the API reference are failing (like to `v2.html#operation/listNodes`) whereas the link to `v2.html` passes. Will be working to resolve.)

**Comment by mikeCRL** [comment]
Updates:

Even though my relative links to subsections of the reference, like to https://deploy-preview-10532--cockroachdb-docs.netlify.app/docs/api/cluster/v2.html#operation/listNodes, work in a web browser, ever since I recently rebased onto master, they are not passing the CI check (while links to https://deploy-preview-10532--cockroachdb-docs.netlify.app/docs/api/cluster/v2.html - sans anchor/hash - do pass).

So I have replaced these with the live-site URLs like https://www.cockroachlabs.com/docs/api/cluster/v2.html#operation/listNodes and the CI now passes. CC @yzdocs as FYI, in case there is anything here that may cause trouble in the future. (Full URLs generally sound like a bad idea because they could pass during CI checks on the branch, but then fail on master after merging, however in this case, the PR doesn't affect those target URLs, so I believe this should be fine.)

Build log excerpt below, for reference. Note that at the time of this check, a link simply to `../api/cluster/v2.html` passed the check. This was not a one-off - I'd also run the build about 10 hours prior with the same code and same result.

```
9:30:47 AM: htmltest started at 01:30:47 on _site
9:30:47 AM: ========================================================================
9:30:48 AM: running in concurrent mode, this is experimental
9:30:59 AM: docs/v21.1/cluster-api.html
9:30:59 AM:   hash does not exist --- docs/v21.1/cluster-api.html --> ../api/cluster/v2.html#operation/health
9:30:59 AM:   hash does not exist --- docs/v21.1/cluster-api.html --> ../api/cluster/v2.html#operation/listNodes
9:30:59 AM:   hash does not exist --- docs/v21.1/cluster-api.html --> ../api/cluster/v2.html#operation/listNodeRanges
9:30:59 AM:   hash does not exist --- docs/v21.1/cluster-api.html --> ../api/cluster/v2.html#operation/listHotRanges
9:30:59 AM:   hash does not exist --- docs/v21.1/cluster-api.html --> ../api/cluster/v2.html#operation/listRange
9:30:59 AM:   hash does not exist --- docs/v21.1/cluster-api.html --> ../api/cluster/v2.html#operation/listSessions
9:30:59 AM:   hash does not exist --- docs/v21.1/cluster-api.html --> ../api/cluster/v2.html#operation/login
9:30:59 AM:   hash does not exist --- docs/v21.1/cluster-api.html --> ../api/cluster/v2.html#operation/logout
9:31:04 AM: ========================================================================
9:31:04 AM: ✘✘✘ failed in 16.935577222s
9:31:04 AM: 8 errors in 3655 documents
9:31:05 AM: ​
9:31:05 AM: ────────────────────────────────────────────────────────────────
9:31:05 AM:   "build.command" failed      
```

File: v21.1/cluster-api.md
Comment (mikeCRL) [reply v21.1/cluster-api.md:32]: @knz Thanks for catching this. It's my mistake; in testing, I see I'm required to specify the _CA cert_ for each call; the /login call and every endpoint after that.

Will it be accurate to say that a requirement is that 'the API client must have a copy of the CA cert used to configure the cluster and other clients'?

Comment (knz) [reply v21.1/cluster-api.md:32]: Why is the CA certificate necessary? what breaks if you don't have it?

Comment (mikeCRL) [reply v21.1/cluster-api.md:32]: Below is what I get in testing - see the API call without, and then with, specifying the CA cert.

From a Slack chat (pasted further below) I thought @itsbilal indicated this was required for all secure calls, though it's possible that there was a misunderstanding. I do know that it was always included in calls during [acceptance testing](https://docs.google.com/document/d/1bj7_DNSLZTvH0SLpWA4DT8SukD0y9DZb6VP3PTdjKNk/edit).

``` bash
ml@mbp ~ % cd crdb/clusters/apitest

ml@mbp apitest % curl -d "username=mike&password=test" -H 'Content-Type: application/x-www-form-urlencoded' --cacert certs/ca.crt https://localhost:8080/api/v2/login/
{"session":"CIGAus6L2ImRCRIQwqYOrikuxijYkyDaARW/5Q=="}%

ml@mbp apitest % TOKEN="CIGAus6L2ImRCRIQwqYOrikuxijYkyDaARW/5Q=="

ml@mbp apitest % curl -H "X-Cockroach-API-Session: $TOKEN" https://localhost:8080/api/v2/sessions/
curl: (60) SSL certificate problem: unable to get local issuer certificate
More details here: https://curl.haxx.se/docs/sslcerts.html

curl failed to verify the legitimacy of the server and therefore could not
establish a secure connection to it. To learn more about this situation and
how to fix it, please visit the web page mentioned above.

ml@mbp apitest % curl -H "X-Cockroach-API-Session: $TOKEN" --cacert certs/ca.crt https://localhost:8080/api/v2/sessions/
{"sessions":[{"node_id":1,"username":"root","client_address":"127.0.0.1:50124","application_name":"$ cockroach sql","active_queries":null,"start":"2021-05-13T14:27:37.223786Z","last_active_query":"SHOW database","id":"Fn6mkOmOo5AAAAAAAAAAAQ==","max_alloc_bytes":952320,"last_active_query_anon":"SHOW database"}],"errors":[],"next":"fDB8MHw="}%
```

```
Mike:
I am going to try to add --cacert to the example curl requests - as it seems virtually required per our earlier chat. Would you agree with doing that? 

Bilal:
sure - though I wonder if that’ll convey its optionality for insecure clusters
I guess you could say curl --cacert <CA cert for secure clusters> <rest of the args>
or curl [--cacert <CA cert for secure clusters>] <rest of the args>
though that makes it less copy-paste-able
```

I can't find the "earlier" transcript referenced there.

Comment (knz) [reply v21.1/cluster-api.md:32]: All right now I understand what is going on. This is necessary because in our currently documented usage, crdb uses a non-standard CA for the HTTP port, and obviously `curl` doesn't know anything about it; so you need to tell it.

However in production usage, we have more and more customers who decide to issue their own HTTP certificates using a public, well-recognized CA. Alternatively, they use a HTTPS load balancer that presents a separate TLS cert to http clients than the one configured inside crdb. In that case, more often than not, `curl` knows about the CA already and there is no need to explain it.

I think it would be important to author a call out to explain these nuances. The CA cert in your examples here are a mere artifact of our (arguably, contrived) security setup in  examples. IT's important to emphasize that end-users should use the parameters that are suitable for their deployment, and that they should seek to determine first if their setup uses a standard CA or not, and only seek out a particular file if their HTTP endpoint uses a non-standard CA (such as the one we build in other doc examples).

Comment (mlewis1) [reply v21.1/cluster-api.md:32]: Got it. Thanks. (We'll have to revisit our use of such security examples and their framing in the docs, as we've touched on before.)

Would this work, for this doc?

> [you will need...]
> 
> * The CA cert used by the cluster or any intermediary proxy server, either in the client's cert store as a trusted certificate authority or as a file manually specified by the HTTP request (for example, using curl's [cacert](https://curl.se/docs/manpage.html#--cacert)).

Comment (mikeCRL) [reply v21.1/cluster-api.md:32]: Got it. Thanks. (We'll have to revisit our use of such security examples and their framing in the docs, as we've touched on before.)

Would this work, for this doc?

> [you will need...]
> 
> * The CA cert used by the cluster or any intermediary proxy server, either in the client's cert store as a trusted certificate authority or as a file manually specified by the HTTP request (for example, using curl's [cacert](https://curl.se/docs/manpage.html#--cacert)).

**Review by mikeCRL — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/docs/10532)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @itsbilal and @taroface)

___
*[v21.1/cluster-api.md, line 9 at r7](https://reviewable.io/reviews/cockroachdb/docs/10532#-M_XqIAX78UUl6EvKeGK:-M_b_St9-3_eYrQQN9H0:b-rxo34t) ([raw file](https://github.com/cockroachdb/docs/blob/8cfdf678cd14329ef27cebc985307ed42eb5ae06/v21.1/cluster-api.md#L9)):*
<details><summary><i>Previously, taroface (Ryan Kuo) wrote…</i></summary><blockquote>

Suggestion: "the same port that listens for HTTP connections"

Also, mention default port (I think it's `8080`)?
</blockquote></details>

Rephrased to convey that in another way. Thanks.
___
*[v21.1/cluster-api.md, line 20 at r7](https://reviewable.io/reviews/cockroachdb/docs/10532#-M_XqqhY5WBr_n4cxgh9:-M_b_aGn4GANSUD9o60Z:b-896fix) ([raw file](https://github.com/cockroachdb/docs/blob/8cfdf678cd14329ef27cebc985307ed42eb5ae06/v21.1/cluster-api.md#L20)):*
<details><summary><i>Previously, taroface (Ryan Kuo) wrote…</i></summary><blockquote>

Suggestion: "Get details on all nodes in the cluster, including..."
</blockquote></details>

Done.
___
*[v21.1/cluster-api.md, line 21 at r7](https://reviewable.io/reviews/cockroachdb/docs/10532#-M_XtVH54Yb4VUCbZc_s:-M_b_b075NmI0iNYY4Pq:b-896fix) ([raw file](https://github.com/cockroachdb/docs/blob/8cfdf678cd14329ef27cebc985307ed42eb5ae06/v21.1/cluster-api.md#L21)):*
<details><summary><i>Previously, taroface (Ryan Kuo) wrote…</i></summary><blockquote>

Suggest "obtain" -> "get" for consistency with the other API descriptions
</blockquote></details>

Done.
___
*[v21.1/cluster-api.md, line 22 at r7](https://reviewable.io/reviews/cockroachdb/docs/10532#-M_Xr5TC4psra7zUpIP9:-M_b_k7W2eyND67zRqYP:b-fbv0ts) ([raw file](https://github.com/cockroachdb/docs/blob/8cfdf678cd14329ef27cebc985307ed42eb5ae06/v21.1/cluster-api.md#L22)):*
<details><summary><i>Previously, taroface (Ryan Kuo) wrote…</i></summary><blockquote>

Suggest "Get details" for consistency with the other API descriptions

Nit: Ideally we could have a more specific threshold for "high number". Not sure if there is a concrete definition of "hot ranges" though.
</blockquote></details>

Done, and  agree about hot ranges. I couldn't find a good definition in the docs. Making a note to follow up with an issue for this.
___
*[v21.1/cluster-api.md, line 23 at r7](https://reviewable.io/reviews/cockroachdb/docs/10532#-M_XrL4u8FR2200KGUbU:-M_b_vFR8OcMsMdvL0xJ:b-896fix) ([raw file](https://github.com/cockroachdb/docs/blob/8cfdf678cd14329ef27cebc985307ed42eb5ae06/v21.1/cluster-api.md#L23)):*
<details><summary><i>Previously, taroface (Ryan Kuo) wrote…</i></summary><blockquote>

Suggestion: "Typically used by Cockroach Labs engineers."
</blockquote></details>

Done.
___
*[v21.1/cluster-api.md, line 24 at r7](https://reviewable.io/reviews/cockroachdb/docs/10532#-M_XrbXwALoU_nqo6ZoV:-M_b_vvk3mnpBhN5Ea5G:b-896fix) ([raw file](https://github.com/cockroachdb/docs/blob/8cfdf678cd14329ef27cebc985307ed42eb5ae06/v21.1/cluster-api.md#L24)):*
<details><summary><i>Previously, taroface (Ryan Kuo) wrote…</i></summary><blockquote>

Suggestion: "Get SQL session details of all current users or a specified user."
</blockquote></details>

Done.
___
*[v21.1/cluster-api.md, line 25 at r7](https://reviewable.io/reviews/cockroachdb/docs/10532#-M_XrmivDIGUnWaIdNxM:-M_ba-Vr45k1_z-VCYEm:b-896fix) ([raw file](https://github.com/cockroachdb/docs/blob/8cfdf678cd14329ef27cebc985307ed42eb5ae06/v21.1/cluster-api.md#L25)):*
<details><summary><i>Previously, taroface (Ryan Kuo) wrote…</i></summary><blockquote>

Should this be `admin`, as in `admin` role? https://www.cockroachlabs.com/docs/v20.2/authorization.html#default-roles
</blockquote></details>

Done.
___
*[v21.1/cluster-api.md, line 34 at r7](https://reviewable.io/reviews/cockroachdb/docs/10532#-M_Xs21-7HPOUCYrMHu9:-M_ba4dc8jpzZ2sDIxic:b-sfk5la) ([raw file](https://github.com/cockroachdb/docs/blob/8cfdf678cd14329ef27cebc985307ed42eb5ae06/v21.1/cluster-api.md#L34)):*
<details><summary><i>Previously, taroface (Ryan Kuo) wrote…</i></summary><blockquote>

Should these be `/health` and `/login` or is this a convention I'm unfamiliar with?
</blockquote></details>

Vestige of something I was trying with the format of all URLs in the doc and where to split the paths to stay 'more true' to URI spec. Meant to change these. Now fixed. Thanks!
___
*[v21.1/cluster-api.md, line 36 at r7](https://reviewable.io/reviews/cockroachdb/docs/10532#-M_Xs5xN2edWm7n1N8IJ:-M_baXCO8maIfOWAkYiD:b-896fix) ([raw file](https://github.com/cockroachdb/docs/blob/8cfdf678cd14329ef27cebc985307ed42eb5ae06/v21.1/cluster-api.md#L36)):*
<details><summary><i>Previously, taroface (Ryan Kuo) wrote…</i></summary><blockquote>

Use backticks on "admin"
</blockquote></details>

Done.
___
*[v21.1/cluster-api.md, line 40 at r7](https://reviewable.io/reviews/cockroachdb/docs/10532#-M_XsCNI5g_XGE8W559I:-M_baXhADsrX0yGWcQtw:b-896fix) ([raw file](https://github.com/cockroachdb/docs/blob/8cfdf678cd14329ef27cebc985307ed42eb5ae06/v21.1/cluster-api.md#L40)):*
<details><summary><i>Previously, taroface (Ryan Kuo) wrote…</i></summary><blockquote>

`/login`?
</blockquote></details>

Done.
___
*[v21.1/cluster-api.md, line 42 at r7](https://reviewable.io/reviews/cockroachdb/docs/10532#-M_XsS39093Fr3JKqYSP:-M_bb0hx3PaMS58azOkH:b810xfn) ([raw file](https://github.com/cockroachdb/docs/blob/8cfdf678cd14329ef27cebc985307ed42eb5ae06/v21.1/cluster-api.md#L42)):*
<details><summary><i>Previously, taroface (Ryan Kuo) wrote…</i></summary><blockquote>

Include `{% include copy-clipboard.html %}` to add the Copy button on any example commands.

Also, replace triple backticks with `~~~` throughout.
</blockquote></details>

Done. (Added the includes in a second follow-up commit, in case you don't see it in Reviewable.)
___
*[v21.1/cluster-api.md, line 58 at r7](https://reviewable.io/reviews/cockroachdb/docs/10532#-M_Xsizr5Sr8PTsgNciR:-M_bbQ4_EAu1I5qrKgnv:b-896fix) ([raw file](https://github.com/cockroachdb/docs/blob/8cfdf678cd14329ef27cebc985307ed42eb5ae06/v21.1/cluster-api.md#L58)):*
<details><summary><i>Previously, taroface (Ryan Kuo) wrote…</i></summary><blockquote>

Since this is an example, maybe `<token>` should go ahead and use the actual token string?

If not, perhaps replace `< >` with `{ }` for style consistency.
</blockquote></details>

Done.
___
*[v21.1/cluster-api.md, line 65 at r7](https://reviewable.io/reviews/cockroachdb/docs/10532#-M_Xsxhh-pRP1jX_MKB-:-M_bbRHzAivnTQuCWxXg:bdsq0ic) ([raw file](https://github.com/cockroachdb/docs/blob/8cfdf678cd14329ef27cebc985307ed42eb5ae06/v21.1/cluster-api.md#L65)):*
<details><summary><i>Previously, taroface (Ryan Kuo) wrote…</i></summary><blockquote>

Suggest breaking into two sentences. 

"...until it is deprecated. Multiple API versions may be provided concurrently." (note typo in "concurrently")
</blockquote></details>

Improved the sentence structure a different way. Thank you.
___
*[v21.1/cluster-api.md, line 67 at r7](https://reviewable.io/reviews/cockroachdb/docs/10532#-M_Xt92o7NySpdKBkguM:-M_bbWjy3nDdcSWQGQtK:b-896fix) ([raw file](https://github.com/cockroachdb/docs/blob/8cfdf678cd14329ef27cebc985307ed42eb5ae06/v21.1/cluster-api.md#L67)):*
<details><summary><i>Previously, taroface (Ryan Kuo) wrote…</i></summary><blockquote>

Suggestion: "would not" -> "will not"
</blockquote></details>

Done.

File: v21.1/cluster-api.md
Comment (mikeCRL) [reply v21.1/cluster-api.md]: Updated the PR to match the above. Please approve if it's now correct/acceptable.

**Review by knz — APPROVED** [review]
> Reviewed 1 of 1 files at r4, 1 of 1 files at r9, 1 of 1 files at r14.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/docs/10532)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @itsbilal and @taroface)

___
*[v21.1/cluster-api.md, line 32 at r7](https://reviewable.io/reviews/cockroachdb/docs/10532#-M_QnnXMtEC1Olgz7w1R-r7-32:-M_er4xj8y4wGEgIgujT:bd0iaob) ([raw file](https://github.com/cockroachdb/docs/blob/8cfdf678cd14329ef27cebc985307ed42eb5ae06/v21.1/cluster-api.md#L32)):*
<details><summary><i>Previously, mikeCRL (Mike Lewis) wrote…</i></summary><blockquote>

Updated the PR to match the above. Please approve if it's now correct/acceptable. 
</blockquote></details>

yes this works thanks

**Pull request merged**
